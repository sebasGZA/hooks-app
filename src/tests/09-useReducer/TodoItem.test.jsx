const { render, screen } = require("@testing-library/react");
const { TodoItem } = require("../../09-useReducer/TodoItem");

describe("Test in <TodoItem />", () => {
  const todo = {
    id: 1,
    description: "Rock of the soul",
    done: false,
  };

  const onRemoveTodoMock = jest.fn();
  const onToggleTodoMock = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("Should make match with snapshot", () => {
    const { container } = render(
      <TodoItem
        key={todo.id}
        todo={todo}
        onToggleTodo={onToggleTodoMock}
        onRemoveTodo={onRemoveTodoMock}
      />
    );

    expect(container).toMatchSnapshot();

    const liElement = screen.getByRole("listitem");
    expect(liElement.className).toBe(
      "list-group-item d-flex justify-content-between"
    );

    const spanElement = screen.getByLabelText("span");
    expect(spanElement.className).toContain("align-self-center");
  });
});
