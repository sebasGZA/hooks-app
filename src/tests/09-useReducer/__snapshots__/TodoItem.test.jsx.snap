// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <TodoItem /> Should make match with snapshot 1`] = `
<div>
  <li
    class="list-group-item d-flex justify-content-between"
  >
    <span
      aria-label="span"
      class="align-self-center "
    >
      Rock of the soul
    </span>
    <button
      class="btn btn-danger"
    >
      Drop
    </button>
  </li>
</div>
`;
